src/contracts/Test/IbexaKernelTestCase.php

```diff
@@ -0,0 +1,178 @@
+use eZ\Publish\SPI\Persistence\TransactionHandler;
+use eZ\Publish\SPI\Tests\Persistence\FixtureImporter;
+use eZ\Publish\SPI\Tests\Persistence\YamlFixture;
+use RuntimeException;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+abstract class IbexaKernelTestCase extends KernelTestCase
+{
+    protected static function getKernelClass(): string
+    {
+        return IbexaTestKernel::class;
+    }
+
+    final protected static function loadSchema(): void
+    {
+        if (!self::$booted) {
+            self::bootKernel();
+        }
+
+        /** @var \eZ\Publish\API\Repository\Tests\LegacySchemaImporter $schemaImporter */
+        $schemaImporter = self::getContainer()->get(LegacySchemaImporter::class);
+        foreach (static::getSchemaFiles() as $schemaFile) {
+            $schemaImporter->importSchema($schemaFile);
+        }
+    }
+
+    /**
+     * @return array<string>
+     */
+    protected static function getSchemaFiles(): iterable
+    {
+        yield self::$kernel->locateResource('@EzPublishCoreBundle/Resources/config/storage/legacy/schema.yaml');
+    }
+
+    final protected static function loadFixtures(): void
+    {
+        if (!self::$booted) {
+            self::bootKernel();
+        }
+
+        /** @var \eZ\Publish\SPI\Tests\Persistence\FixtureImporter $fixtureImporter */
+        $fixtureImporter = self::getContainer()->get(FixtureImporter::class);
+        foreach (static::getFixtures() as $fixture) {
+            $fixtureImporter->import($fixture);
+        }
+
+        static::postLoadFixtures();
+    }
+
+    protected static function postLoadFixtures(): void
+    {
+    }
+
+    /**
+     * @return iterable<\eZ\Publish\SPI\Tests\Persistence\Fixture>
+     */
+    protected static function getFixtures(): iterable
+    {
+        yield new YamlFixture(dirname(__DIR__, 3) . '/eZ/Publish/API/Repository/Tests/_fixtures/Legacy/data/test_data.yaml');
+    }
+
+    /**
+     * @template T of object
+     * @phpstan-param class-string<T> $className
+     *
+     * @return T
+     */
+    final protected static function getServiceByClassName(string $className, ?string $id = null): object
+    {
+        if (!self::$booted) {
+            static::bootKernel();
+        }
+
+        $serviceId = self::getTestServiceId($id, $className);
+        $service = self::getContainer()->get($serviceId);
+        assert(is_object($service) && is_a($service, $className));
+
+        return $service;
+    }
+
+    protected static function getTestServiceId(?string $id, string $className): string
+    {
+        $kernel = self::$kernel;
+        if (!$kernel instanceof IbexaTestKernel) {
+            throw new RuntimeException(sprintf(
+                'Expected %s to be an instance of %s.',
+                get_class($kernel),
+                IbexaTestKernel::class,
+            ));
+        }
+
+        $id = $id ?? $className;
+
+        return $kernel->getAliasServiceId($id);
+    }
```

Comment:
@adamwojs @alongosz `getServiceByClassName()` method has been changed to use the actual Kernel instance, and allow changing how the service ID is acquired by descending classes.

This will make it possible to use this class without `IbexaTestKernel`, and in cases where `createKernel` method is overriden entirely (which is normally the method that calls `getKernelClass()`, and may provide a different way of supplying a Kernel instance).